File: LeetCode/1980/Solution.cpp
class Solution {
public:
    string findDifferentBinaryString(vector<string>& nums) {
        int n = nums.size();
        vector<int> temp;        
        int base = 0;
        
        for(int i=0;i<n;++i) {
            int t = 0;
            int b = 1;
            for(int j=n-1;j>=0;--j) {
                t += (nums[i][j] - '0') * b;
                b *= 2;
            }
            temp.push_back(t);
        }
        
        sort(temp.begin(), temp.end());
        
        for(int i=0;i<n;++i) {
            if(base != temp[i]) {
                break;
            }
            base++;
        }
        
        string ret = "";
        
        while(base) {
            ret = (base % 2 ? "1" : "0") + ret;
            base /= 2;
        }
        
        for(int i=ret.size();i<n;++i) {
            ret = "0" + ret;
        }
        
        return ret;
    }
};

File: LeetCode/1980/Solution.java
class Solution {
    private Set<String> set = new HashSet<>();
    public String findDifferentBinaryString(String[] nums) {
        for(String num : nums) {
            set.add(num);
        }

        return solve("", nums.length);
    }

    private String solve(String temp, int n) {
        if(n == 0) {
            if(set.contains(temp) == false) {
                return temp;
            } else {
                return "";
            }
        }

        String a = solve(temp + "0", n-1);
        if("".equals(a)) {
            return solve(temp + "1", n-1);
        }
        return a;
    }
}
